.github/scripts/check-update-pr-branches.sh

```
@@ -0,0 +1,84 @@
+  echo "--- PR #${PR} を更新中 ---"
+  # 現在の head SHA を取得（失敗時は致命的エラー）
+  if ! HEAD_SHA=$(gh api "repos/${REPO}/pulls/${PR}" --jq '.head.sha' 2>&1); then
+    echo "PR #${PR}: SHA 取得失敗（想定外エラー）"
+    echo "${HEAD_SHA}"
+    exit 1
+  fi
+
+  # ブランチ更新（終了コード + レスポンスボディで分岐）
+  RESPONSE=$(gh api "repos/${REPO}/pulls/${PR}/update-branch" \
+    --method PUT \
+    --field expected_head_sha="${HEAD_SHA}" 2>&1) && {
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)